iree/turbine/kernel/wave/templates/extend_attention.py

```diff
@@ -215,12 +221,14 @@ def first_loop(
             e_delta = tkw.exp2(x_j - m_j)
             e_init = partial_sum * e_delta_max
             d_j = tkw.sum(e_delta, e_init, dim=N_KV)
-            imm_f16 = tkw.cast(e_delta, wave_input_dtype)
```

Comment:
I think we should also keep this as `imm_f16 = tkw.cast(e_delta, wave_input_dtype)` just because we may have bf16 inputs